src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -647,6 +647,23 @@ R"(#include <stdio.h>
                 src = "(bool)(" + src + ")";
                 break;
             }
+            case (ASR::cast_kindType::LogicalToReal) : {
+                src = "(float)(" + src + ")";
+                break;
+            }
+            case (ASR::cast_kindType::RealToLogical) : {
+                src = "(bool)(" + src + ")";
+                break;
+            }
+            case (ASR::cast_kindType::CharacterToLogical) : {
+
+                src = "(bool)(strlen(" + src + ") > 0)";
+                break;
+            }
+            case (ASR::cast_kindType::ComplexToLogical) : {
+                src = "(bool)(" + src + ")";
+                break;
+            }
```

Comment:
Is this case tested in this PR? If not then update `test_builtin_bool.py` to test this complex to logical case.

Reply:
It seems it is already tested by:

```
    c: c32
    c = complex(2, 3)
    assert bool(c)
    c = complex(0, 0)
    assert not bool(c)
    assert not bool(0 + 0j)
```